src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -138,6 +138,9 @@ public WebApplication Build()
             ((IConfigurationBuilder)Configuration).Sources.Clear();
             Configuration.AddConfiguration(_builtApplication.Configuration);
 
+            // Mark the service collection as read-only to prevent future modifications
+            ((WebApplicationServiceCollection)Services).IsReadOnly = true;
```

Comment:
No need to cast, we have a reference to the WebApplicationServiceCollection:

```suggestion
            _services.IsReadOnly = true;
```